inference-engine/src/vpu/graph_transformer/src/frontend/detect_network_batch.cpp

```diff
@@ -82,94 +63,66 @@ void FrontEnd::detectNetworkBatch(
-
-            if (!cmp(layer->type, "DetectionOutput")) {
-                continue;
-            }
-
-            env.log->trace("Found DetectionOutput layer [%s]", layer->name);
-
-            if (layer->outData.empty()) {
-                VPU_THROW_FORMAT("Unsupported layer %s configuration: no outputs", layer->name);
-            }
-
-            // 1. Don't support if DetectionOutput is not the last layer in network
-            if (!ie::getInputTo(layer->outData.front()).empty()) {
-                VPU_THROW_FORMAT("Unsupported layer %s configuration : it is not a network output", layer->name);
-            }
+    const auto layers = network.getFunction()->get_ordered_ops();
+    for (const auto& layer : layers) {
+        if (!ngraph::as_type_ptr<ngraph::op::DetectionOutput>(layer))
+            continue;
+        env.log->trace("Found DetectionOutput layer [%s]", layer->get_name());
 
-            // 2. Don't support if there multiple outputs as well
-            if (outputsInfo.size() != 1) {
-                VPU_THROW_FORMAT("Unsupported network configuration : layer %s must be the only output of the network", layer->name);
-            }
+        VPU_THROW_UNLESS(layer->get_output_size() > 0, "Unsupported layer {} configuration: no outputs", layer->get_name());
 
-            model->attrs().set<bool>("withDetectionOutput", true);
-        }
-    } else {
-        const auto layers = network.getFunction()->get_ops();
-        for (const auto& layer : layers) {
-            if (layer->get_type_name() != std::string("DetectionOutput"))
-                continue;
-
-            env.log->trace("Found DetectionOutput layer [%s]", layer->get_name());
-
-            if (layer->get_output_size() == 0)
-                VPU_THROW_FORMAT("Unsupported layer %s configuration: no outputs", layer->get_name());
-
-            // 1. Don't support if DetectionOutput is not the last layer in network
-            for (const auto& outputHandle : layer->outputs()) {
-                for (const auto& inputHandle : outputHandle.get_target_inputs()) {
-                    auto outNode = inputHandle.get_node();
-                    if (dynamic_cast<::ngraph::opset3::Result *>(outNode)) {
-                        continue;
-                    }
-                    VPU_THROW_FORMAT("Unsupported layer %s configuration : it is not a network output", layer->get_name());
-                }
-            }
-
-            // 2. Don't support if there multiple outputs as well
-            if (outputsInfo.size() != 1) {
-                VPU_THROW_FORMAT("Unsupported network configuration : layer %s must be the only output of the network", layer->get_name());
-            }
-            model->attrs().set<bool>("withDetectionOutput", true);
+        // Don't support if there multiple outputs as well
+        if (outputsInfo.size() != 1) {
+            env.log->trace("Unsupported network configuration : layer %s must be the only output of the network", layer->get_name());
```

Comment:
Please, also mention operation type info

Reply:
Why `env.log->trace` instead of `VPU_THROW_UNLESS`?

Reply:
Type info was added.
The error message was changed.